src/mal_scraper/anime.py

```diff
@@ -103,27 +146,114 @@ def _get_format(soup):
-        logger.warn('No "episodes" tag found.')
-        return None
+        raise MissingTagError('episodes')
 
     episodes_text = pretag.next_sibling.strip().lower()
     if episodes_text == 'unknown':
-        return -1
+        return None
 
     try:
         episodes_number = int(episodes_text)
     except (ValueError, TypeError):  # pragma: no cover
         # MAL probably changed the webpage
-        logger.warn('Unable to convert episodes text "%s" to int.', episodes_text)
-        return None
+        raise ParseError('episodes', 'Unable to convert text "{}" to int'.format(episodes_text))
 
     return episodes_number
+
+
+def _get_airing_status(soup):
+    pretag = soup.find('span', string='Status:')
+    if not pretag:
+        raise MissingTagError('status')
+
+    status_text = pretag.next_sibling.strip().lower()
+    status = {
+        'finished airing': AiringStatus.finished,
+        'currently airing': AiringStatus.ongoing
+    }.get(status_text, None)
+
+    if not status:
+        raise ParseError('status', 'Unable to identify text "{}"'.format(status_text))
+
+    return status
+
+
+def _convert_to_date(text):
+    """Convert MAL text to a datetime stamp.
+
+    Args:
+        text (str): like "Apr 3, 1998"
+
+    Returns:
+        date object or None.
+
+    Raises:
+        ValueError: if the conversion fails
+
+    Issues:
+        - This may be locale dependent
+    """
+    return datetime.strptime(text, '%b %d, %Y').date()
+
+
+def _get_start_date(soup):
+    pretag = soup.find('span', string='Aired:')
+    if not pretag:
+        raise MissingTagError('aired')
+
+    aired_text = pretag.next_sibling.strip()
+    start_text = aired_text.split(' to ')[0]
+
+    try:
+        start_date = _convert_to_date(start_text)
+    except ValueError:
+        raise ParseError('airing start date', 'Cannot process text "%s"'.format(start_text))
+
+    return start_date
+
+
+def _get_end_date(soup):
+    pretag = soup.find('span', string='Aired:')
+    if not pretag:
+        raise MissingTagError('aired')
+
+    aired_text = pretag.next_sibling.strip()
+    end_text = aired_text.split(' to ')[1]
+
+    if end_text == '?':
+        return None
+
+    try:
+        end_date = _convert_to_date(end_text)
+    except ValueError:
+        raise ParseError('airing end date', 'Cannot process text "%s"'.format(start_text))
+
+    return end_date
+
+
+def _get_airing_premiere(soup):
+    pretag = soup.find('span', string='Premiered:')
+    if not pretag:
+        raise MissingTagError('premiered')
+
+    season, year = pretag.find_next('a').string.lower().split(' ')
+
+    if season == 'fall':
+        season = 'autumn'
+    elif season not in ('spring', 'summer', 'autumn', 'winter'):
+        raise ParseError('premiered', 'Unable to identify season "%s"'.format(season))
+
+    try:
+        year = int(year)
+    except (ValueError, TypeError):
+        raise ParseError('premiered', 'Unable to identify year "%s"'.format(year))
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Too many arguments for format string](https://www.codacy.com/app/QasimK/mal-scraper/file/2654365122/issues/source?bid=3298861&fileBranchId=3326344#l257)